Sources/XCTest/XCTestSuite.swift

```diff
@@ -68,3 +68,33 @@ public class XCTestSuite: XCTest {
         tests.append(test)
     }
 }
+
+/// A test suite which is associated with a particular test case class. It will
+/// call `setUp` and `tearDown` on the class itself before and after invoking
+/// all of the test cases making up the class.
+internal class XCTestCaseSuite: XCTestSuite {
+    private let testCaseClass: XCTestCase.Type?
+
+    init(testCaseEntry: XCTestCaseEntry) {
+        let testCaseClass = testCaseEntry.0
+        self.testCaseClass = testCaseClass
+        super.init(name: String(testCaseClass))
+
+        for (testName, testClosure) in testCaseEntry.1 {
+            let testCase = testCaseClass.init(name: testName, testClosure: testClosure)
+            addTest(testCase)
+        }
+    }
+
+    override func setUp() {
+        if let testCaseClass = testCaseClass {
+            testCaseClass.setUp()
+        }
+    }
+
+    override func tearDown() {
+        if let testCaseClass = testCaseClass {
+            testCaseClass.tearDown()
+        }
+    }
+}
```

Comment:
My preference would be to put this in its own file, `XCTestCaseSuite.swift`.

I'd also like to at some point separate the source files in this repository, into directories that make it clear what is public API and what is internal.